src/lib/opts.rs

```diff
@@ -256,6 +269,14 @@ impl Opts {
                 "Same number of read structures should be given as FASTQs"
             );
 
+            // Ensure that if the sample barcodes are to be extracted from the FASTQ header then
+            // no sample barcode segments should be given
+            ensure!(
+                !sample_barcode_in_fastq_header
+                    || read_structures.iter().all(|g| g.sample_barcodes().count() == 0),
+                "Read structures may not contain sample barcode segment(s) when extracting sample barcodes from the FASTQ header"
+            );
```

Comment:
Personally I'd find the following more readable
```suggestion
            if sample_barcode_in_fastq_header {
                ensure!(read_structures.iter().all(|g| g.sample_barcodes().count() == 0),
                    "Read structures may not contain sample barcode segment(s) when extracting sample barcodes from the FASTQ header");
            }
```

Reply:
Done